Issue 93 change hidden tag

waderyan opened the pull request:
@joaomoreno Here are a couple fixes for #94. 

First I changed the hidden tag to be '__' instead of '$'. That was easy. 
The second task I'm having a harder time with and could use your help. An issue we saw with the MP algorithm was javascript extensions being tagged with java. I attempted to fix that by sorounding the regex with '\b' - that solves the "java" issue but introduced a "c++" tagging issue. 

The test is failing now on tagging an extension as "c++". Would love if you could take a look at it. 

Some things I tried. 
- Escaping the regex per instructions [here](http://stackoverflow.com/questions/2593637/how-to-escape-regular-expression-in-javascript) and [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions) - because "+" is a regex character. The function from Mozilla didn't seem to work.

joaomoreno commented:
There are two problems:

1. You must escape the backslashes in `new RegExp('\b' + k + '\b', 'gi')`, thus becoming `new RegExp('\\b' + k + '\\b', 'gi')`.
2. You changed the description in the test to be `This C++ extension [...]`, but there is no `C++` key in the `Keywords` map, so the packaging won't detect it. Should it just be added?

waderyan commented:
Great. Fixed "\\b" issue. 

I'm still having a problem with matching c++. I put it back in the Keywords map and introduced a regex escape method (for the "+" character). Here is what I am seeing. 

![image](https://cloud.githubusercontent.com/assets/2449568/16127175/26ff0a9a-33b0-11e6-8c1e-b13cc5891192.png)

And without word boundary

![image](https://cloud.githubusercontent.com/assets/2449568/16127235/667afdaa-33b0-11e6-9878-0cf400e69423.png)


What am I missing? That should work. I introduced the word boundaries to solve this issue.

![image](https://cloud.githubusercontent.com/assets/2449568/16127220/4dce063a-33b0-11e6-809a-ea178fa5cf8a.png)
 

![image](https://cloud.githubusercontent.com/assets/2449568/16127209/45cb88e0-33b0-11e6-9a46-5fdade009de1.png)

waderyan commented:
Cool - I got the answer on [SO](http://stackoverflow.com/questions/37866277/javascript-regex-matching-word-in-sentence-with-character). 

`+` is not considered a word character.

waderyan commented:
Fixed. Tests passing. Please review :)

joaomoreno commented:
Good job!

joaomoreno commented:
Published in `vsce@1.7.0`
